src/index.js

```diff
@@ -251,121 +340,38 @@ class TerserPlugin {
-              inputSourceMap = map;
-            } else {
-              inputSourceMap = map;
-
-              compilation.warnings.push(
-                new Error(`${file} contains invalid source map`)
-              );
-            }
-          } else {
-            input = asset.source();
-            inputSourceMap = null;
-          }
-
-          // Handling comment extraction
-          let commentsFilename = false;
-
-          if (this.options.extractComments) {
-            commentsFilename =
-              this.options.extractComments.filename || '[file].LICENSE[query]';
-
-            // Todo remove this in next major release
-            if (typeof commentsFilename === 'function') {
-              commentsFilename = commentsFilename.bind(null, file);
-            }
-
-            let query = '';
-            let filename = file;
-
-            const querySplit = filename.indexOf('?');
-
-            if (querySplit >= 0) {
-              query = filename.substr(querySplit);
-              filename = filename.substr(0, querySplit);
-            }
-
-            const lastSlashIndex = filename.lastIndexOf('/');
-
-            const basename =
-              lastSlashIndex === -1
-                ? filename
-                : filename.substr(lastSlashIndex + 1);
-
-            const data = { filename, basename, query };
-
-            commentsFilename = compilation.getPath(commentsFilename, data);
-          }
-
-          if (
-            commentsFilename &&
-            TerserPlugin.hasAsset(commentsFilename, compilation.assets)
-          ) {
-            // Todo make error and stop uglifing in next major release
-            compilation.warnings.push(
-              new Error(
-                `The comment file "${TerserPlugin.removeQueryString(
-                  commentsFilename
-                )}" conflicts with an existing asset, this may lead to code corruption, please use a different name`
-              )
-            );
-          }
-
-          const task = {
-            file,
-            input,
-            inputSourceMap,
-            commentsFilename,
-            extractComments: this.options.extractComments,
-            terserOptions: this.options.terserOptions,
-            minify: this.options.minify,
-          };
-
-          if (this.options.cache) {
-            const defaultCacheKeys = {
-              terser: terserPackageJson.version,
-              // eslint-disable-next-line global-require
-              'terser-webpack-plugin': require('../package.json').version,
-              'terser-webpack-plugin-options': this.options,
-              nodeVersion: process.version,
-              filename: file,
-              contentHash: crypto
-                .createHash('md4')
-                .update(input)
-                .digest('hex'),
-            };
-
-            task.cacheKeys = this.options.cacheKeys(defaultCacheKeys, file);
-          }
+        const defaultCacheKeys = {
```

Comment:
This can be omitted, webpack automatically tracks `buildDependencies`.

We don't track the node.js version, but this shouldn't be necessary.

Reply:
Are you talking about totally `defaultCacheKeys` usage?

Reply:
done